mergify_engine/context.py

```diff
@@ -71,6 +71,12 @@ class PullRequestAttributeError(AttributeError):
     name: str
 
 
+class InstallationCache(typing.TypedDict, total=False):
```

Comment:
why not using a class?

Reply:
I pick the same schema as other caches. Since the cache may cache "None", you need something to differentiate `not-cached` from `None` cached. The none presence of the key means not cached.

Reply:
That's true for the value of the key `team_members`, but the `InstallationCache` itself could be a class. 🤷 I don't see why it needs to be a `TypedDict`.

Reply:
What about something like this: https://github.com/Mergifyio/mergify-engine/pull/3733. I think it's clearer because Cached / NotCached becomes explicit!